backend/siarnaq/api/user/random_avatar.py

```diff
@@ -0,0 +1,70 @@
+import posixpath
+import random
+import uuid
+
+import google.cloud.storage as storage
+from django.conf import settings
+from PIL import Image
+
+from siarnaq.gcloud import titan
+
+
+def generate_avatar(seed):
+    """Return a random avatar as a IMAGE object"""
+    # generate a unique seed
+    random.seed(seed)
+
+    # generate unique rgb
+    rgb1 = tuple(int(random.random() * 255) for _ in range(3))
+    rgb2 = tuple(int(random.random() * 255) for _ in range(3))
+
+    imgsize = settings.GCLOUD_MAX_AVATAR_SIZE  # The size of the image
+
+    avatar = Image.new("RGB", imgsize)  # Create the image
+
+    leftColor = rgb1  # Color at the right
+    rightColor = rgb2  # Color at the left
+
+    for pixel_y in range(imgsize[1]):
+        for pixel_x in range(imgsize[0]):
+
+            # Make it on a scale from 0 to 1
+            dis = float(pixel_x + pixel_y) / (imgsize[0] + imgsize[1])
+
+            # Calculate r, g, and b values
+            r = rightColor[0] * dis + leftColor[0] * (1 - dis)
+            g = rightColor[1] * dis + leftColor[1] * (1 - dis)
+            b = rightColor[2] * dis + leftColor[2] * (1 - dis)
+
+            avatar.putpixel((pixel_x, pixel_y), (int(r), int(g), int(b)))
+
+    return avatar
+
+
+def get_avatar_url(entity):
```

Comment:
The main thing we should definitely do is abstract away the random avatar generation code (the `generate_avatar` function); you've tried to abstract `get_avatar_url` as well, but this seems a bit brittle (e.g. see the comment about hardcoding entities). I would say the simplest solution would be to move the `get_avatar_url` logic back to user and team (copying a bit of logic, which is ok with me for now) and then only keep the `generate_avatar` function here. 

If we really wanted to abstract this logic, it would probably make the most sense to have a custom avatar field with a `get_avatar_url` member function. But that is a bit messy since it changes e.g. the DB model, so probably out of scope for this PR. Pinging @j-mao for any thoughts